### PR TITLE
Draw gyro activity

### PR DESCRIPTION
**Pull request by rbischof**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/SDP/PaintView.java [hunk 1]
```diff
@@ -1,185 +1,70 @@
...
-    private Bitmap bitmap;
-    private MaskFilter embossMask;
-    private MaskFilter blurMask;
-    private Paint bitmapPaint = new Paint(Paint.DITHER_FLAG);
+    private Paint paintc;
+    public int circleRadius;
+    public float circleX;
+    public float circleY;
+    private Path path;
+    public Boolean draw;
```
Comment (Gondolav) [inline app/src/main/java/ch/epfl/sweng/SDP/PaintView.java]: Some of this fields could be private or private final

File: app/src/main/java/ch/epfl/sweng/SDP/PaintView.java [hunk 2]
```diff
@@ -1,185 +1,70 @@
...
     }
 
-    /**
-     * Constructor of our class, initialize the core parameters for our drawing.
-     *
-     * @param context Actual context for drawing.
-     * @param attrs XML attributes of the view.
-     */
-    public PaintView(Context context, AttributeSet attrs) {
+    public PaintView(Context context, AttributeSet attrs){
```
Comment (Gondolav) [inline app/src/main/java/ch/epfl/sweng/SDP/PaintView.java]: The javadoc could remain

File: app/src/androidTest/java/ch/epfl/sweng/SDP/DrawingActivityTest.java [hunk 3]
```diff
@@ -24,4 +24,23 @@
     public void testCanvas() {
         onView(withId(R.id.paintView)).perform(click());
     }
+
+    @Test
+    public void tesDrawToggleIsClickable() {
+        onView(withId(R.id.fly_or_draw)).perform(click());
+        onView(withId(R.id.fly_or_draw)).check(matches(isClickable()));
```
Comment (LudoHoff) [inline app/src/androidTest/java/ch/epfl/sweng/SDP/DrawingActivityTest.java]: "onView(withId(R.id.fly_or_draw)).perform(click())" is useless, you can delete it

File: app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java [hunk 4]
```diff
@@ -1,103 +1,108 @@
...
+        handler = new Handler(){
+           @Override
+           public void handleMessage(Message message){
+               paintView.invalidate();
+           }
+        };
+    }
 
-        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+    public void fly_or_draw(View view){
```
[the remark this answers is not in the document]
Comment (LudoHoff) [reply app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java]: A javadoc is also required

File: app/src/main/java/ch/epfl/sweng/SDP/PaintView.java [hunk 5]
```diff
@@ -1,185 +1,70 @@
...
+        circleX = 0;
+        circleY = 0;
+        draw = false;
+        path = new Path();
+        path.moveTo(circleX, circleY);
     }
 
-    /**
-     * Clear everything that is drawn.
-     */
```
Comment (LudoHoff) [inline app/src/main/java/ch/epfl/sweng/SDP/PaintView.java]: This one should remain too

File: app/src/main/java/ch/epfl/sweng/SDP/PaintView.java [hunk 6]
```diff
@@ -1,185 +1,70 @@
...
-                touchMove(xx, yy);
-                invalidate();
-                break;
-            case MotionEvent.ACTION_UP :
-                touchUp();
-                invalidate();
-                break;
-            default:
+    public void onDraw(Canvas canvas){
+        canvas.save();
```
Comment (LudoHoff) [inline app/src/main/java/ch/epfl/sweng/SDP/PaintView.java]: Missing javadoc

File: app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java [hunk 7]
```diff
@@ -1,103 +1,108 @@
...
-    private float[] accelerometerReading = new float[3];
-    private float[] magnetometerReading = new float[3];
 
-    private float[] rotationMatrix = new float[9];
-    private float[] orientationAngles = new float[3];
+    private static final String TAG = "DrawingActivity";
+    private int speed;
+    private Point size;
+    private Handler handler;
+    private SensorManager sensorManager;
```
[the remark this answers is not in the document]
Comment (rbischof) [reply app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java]: SensorManager, Handler final cause app to crash.
Speed will be modifiable in future.

File: app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java [hunk 8]
```diff
@@ -1,103 +1,108 @@
...
-import android.util.DisplayMetrics;
 import android.util.Log;
+import android.view.Display;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.ToggleButton;
 
-public class DrawingActivity extends AppCompatActivity implements SensorEventListener {
 
-    private static final String TAG = "DrawingActivity";
```
[the remark this answers is not in the document]
Comment (rbischof) [reply app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java]: There is a tag, but different location.

File: app/src/androidTest/java/ch/epfl/sweng/SDP/DrawingActivityTest.java [hunk 9]
```diff
@@ -24,4 +24,23 @@
     public void testCanvas() {
         onView(withId(R.id.paintView)).perform(click());
     }
+
+    @Test
+    public void tesDrawToggleIsClickable() {
```
Comment (Claudio5) [inline app/src/androidTest/java/ch/epfl/sweng/SDP/DrawingActivityTest.java]: Add the ending 't' for test :)

File: app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java [hunk 10]
```diff
@@ -1,103 +1,108 @@
...
+    private SensorManager sensorManager;
+    ToggleButton fly_draw;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_drawing);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+
+        speed = 5;
```
Comment (Claudio5) [inline app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java]: Make the 5 a constant

File: app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java [hunk 11]
```diff
@@ -1,103 +1,108 @@
...
-    private float[] magnetometerReading = new float[3];
 
-    private float[] rotationMatrix = new float[9];
-    private float[] orientationAngles = new float[3];
+    private static final String TAG = "DrawingActivity";
+    private int speed;
+    private Point size;
+    private Handler handler;
+    private SensorManager sensorManager;
+    ToggleButton fly_draw;
```
Comment (Claudio5) [inline app/src/main/java/ch/epfl/sweng/SDP/DrawingActivity.java]: Use flyDraw to keep the same convention for all our code

**Pull request closed without merging**